File: src/game/tools/rpc/output/wrap/HeroCommentDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type HeroCommentDataWrap struct {
	
	CommentID int32 // 评论ID
	HeroID int32 // 英雄ID
	UserID int64 // 用户ID
	UserNick string // 用户昵称
	Content string // 内容
	LikeNum int32 // 点赞数
	CommentTime int64 // 评论时间
	IsLiked bool // 是否已经点赞
	rpc.Wrapper
}

func (w *HeroCommentDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.CommentID = pck.PopInt32(); 
	w.HeroID = pck.PopInt32(); 
	w.UserID = pck.PopInt64(); 
	w.UserNick = pck.PopString(); 
	w.Content = pck.PopString(); 
	w.LikeNum = pck.PopInt32(); 
	w.CommentTime = pck.PopInt64(); 
	w.IsLiked = pck.PopBool(); 
	return w
}

func (w *HeroCommentDataWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt32(w.CommentID) 
	pck.PutInt32(w.HeroID) 
	pck.PutInt64(w.UserID) 
	pck.PutString(w.UserNick) 
	pck.PutString(w.Content) 
	pck.PutInt32(w.LikeNum) 
	pck.PutInt64(w.CommentTime) 
	pck.PutBool(w.IsLiked) 
}

File: src/game/tools/rpc/output/wrap/HeroCommentDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class HeroCommentDataWrap : IWrapper
	{	
		public int CommentID; // 评论ID
		public int HeroID; // 英雄ID
		public long UserID; // 用户ID
		public string UserNick; // 用户昵称
		public string Content; // 内容
		public int LikeNum; // 点赞数
		public long CommentTime; // 评论时间
		public bool IsLiked; // 是否已经点赞
		
		public void Decode(Packet pck)
		{	
			CommentID = pck.GetInt(); 
			HeroID = pck.GetInt(); 
			UserID = pck.GetLong(); 
			UserNick = pck.GetString(); 
			Content = pck.GetString(); 
			LikeNum = pck.GetInt(); 
			CommentTime = pck.GetLong(); 
			IsLiked = pck.GetBool(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutInt(CommentID); 
        	pck.PutInt(HeroID); 
        	pck.PutLong(UserID); 
        	pck.PutString(UserNick); 
        	pck.PutString(Content); 
        	pck.PutInt(LikeNum); 
        	pck.PutLong(CommentTime); 
        	pck.PutBool(IsLiked); 
        }
	}
}